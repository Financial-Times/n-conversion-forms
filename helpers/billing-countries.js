import { billingCountries } from 'n-common-static-data';

export { billingCountries };
