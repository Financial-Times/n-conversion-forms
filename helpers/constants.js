const printRegions = {
	cemeaV1: 'CEMEA_V1',
	cemeaV2: 'CEMEA_V2',
	apac: 'APAC',
};

const addressLine1AutoComplete = 'address-line1';
const addressLine2AutoComplete = 'address-line2';
const addressLine3AutoComplete = 'address-line3';

module.exports = {
	printRegions,
	addressLine1AutoComplete,
	addressLine2AutoComplete,
	addressLine3AutoComplete,
};
