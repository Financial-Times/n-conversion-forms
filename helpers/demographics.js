const { demographics } = require('n-common-static-data');

export { demographics };
