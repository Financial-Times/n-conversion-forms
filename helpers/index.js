module.exports = {
	'ncf-countries': require('./ncf-countries'),
	'ncf-common-data': require('./ncf-common-data'),
	cemeaV1ISO: require('./supportedCountries').cemeaV1ISO,
	cemeaV2ISO: require('./supportedCountries').cemeaV2ISO,
	apacISO: require('./supportedCountries').apacISO,
	identifyFTShippingZone: require('./supportedCountries')
		.identifyFTShippingZone,
	countriesSupported: require('./supportedCountries').countriesSupported,
	countriesSupportedISO: require('./supportedCountries').countriesSupportedISO,
	supportedPostcodeValidators: require('./supportedPostcodeValidators')
		.supportedPostcodeValidators,
	allSupportedPostcodeExamples: require('./supportedPostcodeExamples')
		.allSupportedPostcodeExamples,
	demographics: require('./demographics').demographics,
	billingCountries: require('./billing-countries').billingCountries,
};
