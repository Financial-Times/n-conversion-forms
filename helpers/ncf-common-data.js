const _get = require('lodash.get');
const COMMON_DATA = require('n-common-static-data');

module.exports = function ({ hash, fn }) {
	const context = Object.assign({}, { [hash.export]: _get(COMMON_DATA, hash.import) }, this);
	return fn(context);
};
