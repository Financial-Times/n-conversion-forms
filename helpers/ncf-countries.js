const { countries } = require('n-common-static-data').billingCountries;

const MINIMUM_TO_GROUP = 50;
const MINIMUM_TO_SHOW_GROUPS = 2;

module.exports = function ({ hash = {}, fn }) {
	const filter = Array.isArray(hash.filterList) ? hash.filterList : false;
	const value = hash.value;

	let data = countries;

	// Only show countries in the filter
	if (filter) {
		data = data.filter(item => filter.includes(item.code));
	}

	// Select the country
	if (value) {
		data = data.map(item => {
			item.selected = item.code === value;
			return item;
		});
	}

	// Group countries
	if (data.length >= MINIMUM_TO_GROUP) {
		data = groupCountries(data);
	}

	const context = Object.assign({ countries: data }, this);
	return fn(context);
};

/**
 * Produce a frequently used option group
 * @param {Array} countries
 * @return {Array}
 */
function groupCountries (countries) {
	const frequentlyUsed = ['GBR', 'USA', 'JPN', 'FRA', 'CAN'];

	const frequentlyUsedCountries = countries
		.filter(item => frequentlyUsed.includes(item.code))
		.map(item => Object.assign({}, item))
		.sort((a, b) => frequentlyUsed.indexOf(a.code) - frequentlyUsed.indexOf(b.code));

	// Not worth showing return standard countries list
	if (frequentlyUsedCountries.length < MINIMUM_TO_SHOW_GROUPS) {
		return countries;
	}

	// If selected country is in frequently used only leave that one selected
	// otherwise the user is shown the lowest selected option
	const selected = frequentlyUsedCountries.find(item => item.selected);
	if (selected) {
		countries.forEach(item => item.selected = false);
	}

	return [
		{ label: 'Frequently Used', countries: frequentlyUsedCountries },
		{ label: 'Alphabetical', countries }
	];
}
