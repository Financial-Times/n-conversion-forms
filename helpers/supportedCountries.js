const supportedCountriesMasterList = {
	cemeaV1: {
		AUT: 'Austria',
		BEL: 'Belgium',
		BGR: 'Bulgaria',
		HRV: 'Croatia',
		CZE: 'Czechia',
		DNK: 'Denmark',
		FIN: 'Finland',
		DEU: 'Germany',
		GRC: 'Greece',
		HUN: 'Hungary',
		ISL: 'Iceland',
		ITA: 'Italy',
		LIE: 'Liechtenstein',
		LTU: 'Lithuania',
		NLD: 'Netherlands',
		NOR: 'Norway',
		POL: 'Poland',
		PRT: 'Portugal',
		QAT: 'Qatar',
		ROU: 'Romania',
		RUS: 'Russian Federation',
		SVK: 'Slovakia',
		SVN: 'Slovenia',
		ZAF: 'South Africa',
		ESP: 'Spain',
		SWE: 'Sweden',
		CHE: 'Switzerland'
	},
	cemeaV2: { FRA: 'France', LUX: 'Luxembourg', MCO: 'Monaco' },
	apac: {
		CHN: 'China',
		HKG: 'Hong Kong',
		IND: 'India',
		IDN: 'Indonesia',
		JPN: 'Japan',
		KOR: 'Korea, Republic of',
		MYS: 'Malaysia',
		PHL: 'Philippines',
		SGP: 'Singapore',
		TWN: 'Taiwan, Province of China',
		THA: 'Thailand'
	},
	GBR: 'United Kingdom',
	USA: 'United States',
	CAN: 'Canada'
};


const cemeaV1ISO = Object.keys(supportedCountriesMasterList.cemeaV1);
const cemeaV2ISO = Object.keys(supportedCountriesMasterList.cemeaV2);
const apacISO = Object.keys(supportedCountriesMasterList.apac);
const countriesSupported = Object.assign({}, ...(function _flatten (o) { return [].concat(...Object.keys(o).map(k => typeof o[k] === 'object' ? _flatten(o[k]) : ({[k]: o[k]})));}(supportedCountriesMasterList)));


module.exports = { cemeaV1ISO, cemeaV2ISO, apacISO, countriesSupported };
