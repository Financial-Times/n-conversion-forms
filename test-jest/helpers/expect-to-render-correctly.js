import ReactDOMServer from 'react-dom/server';

export const expectToRenderCorrectly = {
	toRenderCorrectly(reactComponent, props) {
		// Render the React component with props.
		const ReactComponentRendered = ReactDOMServer.renderToStaticMarkup(
			reactComponent(props)
		);

		// eslint-disable-next-line jest/no-standalone-expect
		expect(ReactComponentRendered).toMatchSnapshot();

		// This point is reached when the above assertions were successful.
		// The test should therefore always pass, that means it needs to be
		// `true` when used normally, and `false` when `.not` was used.
		return {
			pass: !this.isNot,
		};
	},
};
