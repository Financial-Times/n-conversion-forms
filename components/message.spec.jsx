import { Message } from './index';
import { expectToRenderCorrectly } from '../test-jest/helpers/expect-to-render-correctly';

expect.extend(expectToRenderCorrectly);

describe('Message', () => {
	it('render a message', () => {
		const props = {
			message: 'My message in a bottle',
		};

		expect(Message).toRenderCorrectly(props);
	});

	it('can render a title', () => {
		const props = {
			title: 'Reggatta de Blanc',
			message: 'My message in a bottle',
		};

		expect(Message).toRenderCorrectly(props);
	});

	it('can render additional information', () => {
		const props = {
			title: 'Reggatta de Blanc',
			message: 'My message in a bottle',
			additional: ['Sting', 'Steward Copeland', 'Andy Summers'],
		};

		expect(Message).toRenderCorrectly(props);
	});

	it('can render some actions as primary', () => {
		const props = {
			title: 'Reggatta de Blanc',
			message: 'My message in a bottle',
			actions: [
				{
					text: 'Listen on Spotify',
					link: 'https://open.spotify.com/album/2EpuND32cO7CX0gXZl2NB6',
				},
			],
		};

		expect(Message).toRenderCorrectly(props);
	});

	it('can render some actions as secondary', () => {
		const props = {
			title: 'Reggatta de Blanc',
			message: 'My message in a bottle',
			actions: [
				{
					text: 'Listen on Spotify',
					link: 'https://open.spotify.com/album/2EpuND32cO7CX0gXZl2NB6',
					isSecondary: true,
				},
			],
		};

		expect(Message).toRenderCorrectly(props);
	});

	it('can render a message as a Notice', () => {
		const props = {
			message: 'My message in a bottle',
			isNotice: true,
		};

		expect(Message).toRenderCorrectly(props);
	});

	it('can render a message as an Error', () => {
		const props = {
			message: 'My message in a bottle',
			isError: true,
		};

		expect(Message).toRenderCorrectly(props);
	});

	it('can render a message as a Success', () => {
		const props = {
			message: 'My message in a bottle',
			isSuccess: true,
		};

		expect(Message).toRenderCorrectly(props);
	});

	it('can render a message as an Inform', () => {
		const props = {
			message: 'My message in a bottle',
			isInform: true,
		};

		expect(Message).toRenderCorrectly(props);
	});

	it('can add a data attribute name', () => {
		const props = {
			message: 'My message in a bottle',
			name: 'The Police best album ever',
		};

		expect(Message).toRenderCorrectly(props);
	});
});
