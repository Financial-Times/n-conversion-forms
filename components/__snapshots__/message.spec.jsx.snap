// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message can add a data attribute name 1`] = `
<div class="ncf__message o-forms o-forms--wide"
     data-message-name="The Police best album ever"
>
  <div class="o-message o-message--alert-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can add a data attribute name 2`] = `
<div class="ncf__message o-forms o-forms--wide"
     data-message-name="The Police best album ever"
>
  <div class="o-message o-message--alert-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render a message as a Notice 1`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--notice-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render a message as a Notice 2`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--notice-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render a message as a Success 1`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--success"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render a message as a Success 2`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--success"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render a message as an Error 1`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--error"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render a message as an Error 2`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--error"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render a message as an Inform 1`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--inform"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render a message as an Inform 2`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--inform"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render a title 1`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-highlight">
            Reggatta de Blanc
          </span>
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render a title 2`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-highlight">
            Reggatta de Blanc
          </span>
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render additional information 1`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-highlight">
            Reggatta de Blanc
          </span>
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
        <p class="o-message__content--additional">
          Sting
        </p>
        <p class="o-message__content--additional">
          Steward Copeland
        </p>
        <p class="o-message__content--additional">
          Andy Summers
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render additional information 2`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-highlight">
            Reggatta de Blanc
          </span>
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
        <p class="o-message__content--additional">
          Sting
        </p>
        <p class="o-message__content--additional">
          Steward Copeland
        </p>
        <p class="o-message__content--additional">
          Andy Summers
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render some actions as primary 1`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-highlight">
            Reggatta de Blanc
          </span>
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
        <div class="o-message__actions">
          <a href="https://open.spotify.com/album/2EpuND32cO7CX0gXZl2NB6"
             class="o-message__actions__primary"
          >
            Listen on Spotify
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render some actions as primary 2`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-highlight">
            Reggatta de Blanc
          </span>
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
        <div class="o-message__actions">
          <a href="https://open.spotify.com/album/2EpuND32cO7CX0gXZl2NB6"
             class="o-message__actions__primary"
          >
            Listen on Spotify
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render some actions as secondary 1`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-highlight">
            Reggatta de Blanc
          </span>
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
        <div class="o-message__actions">
          <a href="https://open.spotify.com/album/2EpuND32cO7CX0gXZl2NB6"
             class="o-message__actions__secondary"
          >
            Listen on Spotify
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message can render some actions as secondary 2`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-highlight">
            Reggatta de Blanc
          </span>
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
        <div class="o-message__actions">
          <a href="https://open.spotify.com/album/2EpuND32cO7CX0gXZl2NB6"
             class="o-message__actions__secondary"
          >
            Listen on Spotify
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message render a message 1`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message render a message 2`] = `
<div class="ncf__message o-forms o-forms--wide">
  <div class="o-message o-message--alert-inner o-message--neutral"
       data-o-component="o-message"
  >
    <div class="o-message__container">
      <div class="o-message__content">
        <p class="o-message__content-main">
          <span class="o-message__content-detail">
            My message in a bottle
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
